src/server_manager/web_app/gcp_account.ts

```diff
@@ -12,16 +12,309 @@
+    let createProjectOperation = null;
+    while (!createProjectOperation?.done) {
+      await sleep(2 * 1000);
+      createProjectOperation =
+          await this.apiClient.resourceManagerOperationGet(createProjectResponse.name);
+    }
+    if (createProjectOperation.error) {
+      // TODO: Throw error. The project wasn't created so we should have nothing to delete.
+    }
+
+    await this.configureProject(projectId, billingAccountId);
+
+    return {
+      id: projectId,
+      name: GcpAccount.OUTLINE_PROJECT_NAME,
+    };
+  }
+
+  async isProjectHealthy(projectId: string): Promise<boolean> {
+    const projectBillingInfo = await this.apiClient.getProjectBillingInfo(projectId);
+    if (!projectBillingInfo.billingEnabled) {
+      return false;
+    }
+
+    const listEnabledServicesResponse = await this.apiClient.listEnabledServices(projectId);
+    for (const requiredService of GcpAccount.REQUIRED_GCP_SERVICES) {
+      const found = listEnabledServicesResponse.services.find(
+          service => service.config.name === requiredService);
+      if (!found) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  async repairProject(projectId: string, billingAccountId: string): Promise<void> {
+    return await this.configureProject(projectId, billingAccountId);
+  }
+
+  /** @see {@link Account#listBillingAccounts}. */
+  async listBillingAccounts(): Promise<BillingAccount[]> {
+    const response = await this.apiClient.listBillingAccounts();
+    if (response.billingAccounts?.length > 0) {
+      return response.billingAccounts.map(billingAccount => {
+        return {
+          id: billingAccount.name.substring(billingAccount.name.lastIndexOf('/') + 1),
+          name: billingAccount.displayName,
+        };
+      });
+    }
+    return [];
+  }
+
+  private async createInstance(projectId: string, name: string, zoneId: string):
+      Promise<gcp_api.Instance> {
+    // Configure Outline firewall
+    const getFirewallResponse =
+        await this.apiClient.listFirewalls(projectId, GcpAccount.OUTLINE_FIREWALL_NAME);
+    let createFirewallOperation = null;
+    if (!getFirewallResponse?.items || getFirewallResponse?.items?.length === 0) {
+      const createFirewallData = this.makeCreateFirewallRequestData(name);
+      createFirewallOperation = await this.apiClient.createFirewall(projectId, createFirewallData);
+      createFirewallOperation = await this.apiClient.computeEngineOperationGlobalWait(
+          projectId, createFirewallOperation.name);
+      if (createFirewallOperation.error?.errors) {
+        // TODO: Throw error.
+      }
+    }
+
+    // Create VM instance
+    const createInstanceData = this.makeCreateInstanceRequestData(name, zoneId);
+    let createInstanceOperation =
+        await this.apiClient.createInstance(projectId, zoneId, createInstanceData);
+    createInstanceOperation = await this.apiClient.computeEngineOperationZoneWait(
```

Comment:
TODO: Move the wait to the createInstance call

Reply:
I don't see the change. Did you push it to the right branch?

Reply:
Can you check again? It in this diff: https://github.com/Jigsaw-Code/outline-server/pull/864/commits/0941352b04a9d3a37db8ac7ac6913eeac6925815

I ended up implementing the TODO.